### DISCUSSION TITLE
How to use expand correctly

### DISCUSSION THREAD

**Discussion by MuslemRahimi** [pocketbase/pocketbase#3690] [Q&A] [answered]
Hi

I have the issue where i want to add specific user data into my output result.

Currently I am doing it in a very noob way:
```
const output = await pb.collection('comments').getList(1,500,{
			filter: `post="${params.postId}"`,
		}) ;


		let getUsername;
		for (let item of output.items) {

			getUsername = await pb.collection("users").getOne(item.user);


			item.username = getUsername.username;
			item.userKarma = getUsername.karma;
			item.userCollectionId = getUsername.collectionId;
			item.avatar = getUsername.avatar;
		}


        return output
		 
```

I never really understood how expand works but i guess it would make sense here to apply it right?

**Comment by ganigeorgiev** [comment, marked as answer]
Pass the relation field name as `expand` value. In your case that should look like:

```js
const output = await pb.collection('comments').getList(1,500,{
    filter: `post="${params.postId}"`,
    expand: "user"
});
```

Each `output.items` should have an `expand.user` object (_if the request satisfies the View API rule of the "user" related collection_).
